app/controllers/api/exchanges_products_controller.rb

```diff
@@ -0,0 +1,134 @@
+module Api
+  class ExchangesProductsController < Api::BaseController
+    skip_authorization_check only: [:index, :show]
+
+    # Lists products for an Enterprise in an Order Cycle
+    #   This is the same as index below but when the Exchange doesn't exist yet
+    #
+    #   Parameters are: enterprise_id, order_cycle_id and incoming
+    #     order_cycle_id is optional if for incoming
+    def show
+      load_data_from_params
+
+      render_products
+    end
```

Comment:
this action should be index as well, but... 
the index endpoint corresponds to api/exchanges/{exchange_id}/products
but when there's no exchange_id yet, we call api/exchanges/products and that is sent to the action show of the exchanges resource (not the exchange_products resource).
To redirect to this controller I did this in the routes:
`resources :exchanges, only: [:show], controller: 'exchanges_products' do`
This make api/exchanges/products redirect to this show action in the ExchangeProducts controller.

This is not good, is it better to have this a new ExchangesController? Is there a way to redirect this URL (api/exchanges/products) also to the index action in the ExchangesProductsController?

Reply:
this is not fixed :tada: